hawkular-command-gateway/hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusEndpointProcessors.java

```diff
@@ -222,8 +221,8 @@ protected void onBasicMessage(BasicMessageWithExtraData<BasicMessage> messageWit
     @Inject
     private BusCommandContextFactory commandContextFactory;
 
-    @Resource(name = Constants.CONNECTION_FACTORY_JNDI)
-    private ConnectionFactory connectionFactory;
+    @Inject
+    private BusConnectionFactoryProvider connectionFactoryProvider;
```

Comment:
`Instance` sounds as a good idea, that I'd gladly implement if I only had time for running Agent and Inventory itests enough times to make sure that it works.